src/shared/lib/URLWrapper.ts

```diff
@@ -39,6 +52,23 @@ export default class URLWrapper<QueryParamsType> {
         this.routing.updateRoute(query as any);
     }
 
+    private setProperty(property:Property<QueryParamsType>, query:QueryParamsType){
+        this._setPropertyAndHandleUndefined(property, query[property.name]);
+        // if it's still undefined, then check aliases
+        if (this.query[property.name] === undefined && property.aliases && property.aliases.length) {
+            for (const alias of property.aliases) {
+                this._setPropertyAndHandleUndefined(property, query[alias]);
```

Comment:
I think it would be most idiomatic to have the helper return boolean depending whether it successfully set a value. then you can do if (this.trySyncProperty(..)) break;

a little cleaner I think